tests/infrastructure/test_resource_leaks.py

```
@@ -0,0 +1,309 @@
+        aws_provider.ssh_client = mock_ssh
+
+        # Mock _ssh_exec to return a valid remote path
+        aws_provider._ssh_exec = AsyncMock(return_value=(0, "/home/ubuntu/.mcpbr_run_001\n", ""))
+
+        with pytest.raises((IOError, FileNotFoundError, RuntimeError)):
+            await aws_provider._download_results()
+
+        # SFTP must have been closed despite the error
+        mock_sftp.close.assert_called()
+
+    async def test_temp_file_cleaned_on_sftp_error(self, aws_provider, tmp_path) -> None:
```

Ruff (0.14.14)

[warning] Unused method argument: `tmp_path`

(ARG002)